datafusion/sql/src/expr/value.rs

```diff
@@ -151,16 +151,51 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         let data_types: HashSet<DataType> =
             values.iter().map(|e| e.get_datatype()).collect();
 
-        if data_types.is_empty() {
-            Ok(lit(ScalarValue::new_list(None, DataType::Utf8)))
-        } else if data_types.len() > 1 {
-            Err(DataFusionError::NotImplemented(format!(
-                "Arrays with different types are not supported: {data_types:?}",
-            )))
-        } else {
-            let data_type = values[0].get_datatype();
+        match data_types.len() {
+            0 => Ok(lit(ScalarValue::new_list(None, DataType::Utf8))),
+            1 => {
+                let data_type = values[0].get_datatype();
+
+                Ok(lit(ScalarValue::new_list(Some(values), data_type)))
+            }
+            // Check whether one of them is null
+            2 => {
```

Comment:
Why does this stop at 2? Why not handle 3 or more different data types?

Reply:
I didn't sure what is the case for >2.
If you have int, float, and null in arrays, int, and float should be coerced before null handling, so I think only null and one data type are expected.

Reply:
> I didn't sure what is the case for >2. If you have int, float, and null in arrays, int, and float should be coerced before null handling, so I think only null and one data type are expected.

False statement